packages/contracts/contracts/L2/ValidatorRewardVault.sol

```diff
@@ -11,10 +14,271 @@ import { Semver } from "../universal/Semver.sol";
+     */
+    function releasePending() external {
+        bool released = tryRelease();
+        require(
+            released,
+            "ValidatorRewardVault: there are no releasable pending rewards"
+        );
+    }
+
+    /**
+     * @notice Disable the existing withdraw function.
+     */
+    function withdraw() external pure override {
+        revert("ValidatorRewardVault: cannot be withdrawn with this function");
+    }
+
+    /**
+     * @notice Withdraws all of the given recipient's balance.
+     *         Reverts if the balance is less than the minimum withdrawal amount.
+     * 
+     * @param _recipient Address of the recipient.
+     */
+    function withdraw(address _recipient) external {
+        uint256 balance = balances[_recipient];
+
+        require(
+            balance >= MIN_WITHDRAWAL_AMOUNT,
+            "ValidatorRewardVault: withdrawal amount must be greater than minimum withdrawal amount"
+        );
+
+        balances[_recipient] = 0;
+        totalProcessed += balance;
+
+        emit Withdrawal(balance, _recipient, msg.sender);
+
+        L2StandardBridge(payable(Predeploys.L2_STANDARD_BRIDGE)).bridgeETHTo{ value: balance }(
+            _recipient,
+            WITHDRAWAL_MIN_GAS,
+            bytes("")
+        );
+    }
+
+    /**
+     * @notice Attempts to release a pending reward, and increases recipient's balances.
+     *         If there are no releasable rewards, this function ends quietly.
+     */
+    function tryRelease() internal returns (bool) {
+        uint256 outputIndex = nextReleaseOutputIndex;
+
+        Types.PendingReward storage reward = pendingRewards[outputIndex];
+        if (reward.recipient != address(0) && block.timestamp >= reward.releaseAt) {
+            balances[reward.recipient] += reward.amount;
+            totalPendingReward -= reward.amount;
+
+            emit PendingRewardReleased(
+                outputIndex,
+                reward.recipient,
+                reward.amount,
+                block.timestamp
+            );
+
+            delete pendingRewards[outputIndex];
+            nextReleaseOutputIndex++;
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Returns whether a pending reward corresponding to the given output index exists.
+     * 
+     * @param _outputIndex Index of the L2 checkpoint output index.
+     * 
+     * @return Whether a pending reward corresponding to the given output index exists.
+     */
+    function isExistingPending(uint256 _outputIndex) internal view returns (bool) {
+        return pendingRewards[_outputIndex].recipient != address(0);
+    }
+
+    /**
+     * @notice Determines the reward the recipient can receive at the current time.
+     * 
+     * @return The reward the recipient can receive.
+     */
+    function determineRewardAmount() internal view returns (uint256) {
+        return address(this).balance - totalPendingReward;
+    }
```

Comment:
I think the sum of `balances` (not withdrawn, so included in `address(this).balance`) should be subtracted as well